### PR TITLE
Setup documentation and corresponding github action

### LINKED ISSUE #53: Create unified docs page and host on GitHub pages

### PR DESCRIPTION
**Pull request by wmaxey**
## Description

closes #53

This PR allows building our documentation.

Currently CUB has some failures, these can be ignored for the time being. It seems that Doxygen/Sphinx does its best to create something useable even if there's a build failure.

Libcudacxx/Thrust required a bit of fenagling to point to the right base URL.

There is a top level CCCL build that just indexes the libraries... We can make this better later. 

You can see the output of the latest commit here: https://wmaxey.github.io/cccl/

## Checklist

- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes. (IT IS THE DOCUMENTATION)
- [ ] Make old libraries landing pages point to CCCL. Possibly redirecting to correct page if able.

### CODE REVIEW

File: .github/workflows/build-docs.yml [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+        uses: actions/jekyll-build-pages@v1
+        with:
+          source: ./docs/jekyll
+          destination: ./_site
+
+      # CUB
+      - name: Build CUB docs
+        run: |
+          bash ./docs/build_docs.bash "cccl:docs" /cccl/cub/docs/gen_docs.bash
+          sudo mkdir -p ./_site/cub
```
Comment (wmaxey) [inline .github/workflows/build-docs.yml:49]: I don't know why, but I couldn't figure out the permissions for the output directory. We need `sudo` to allow writing into it.

File: docs/readme.md [hunk 2]
```diff
@@ -0,0 +1,23 @@
+# Building docs
+
+Most docs can be *prepared* locally, but the build process is partially invested in the github workflows.
+
+## First steps
+
+Prepare docker container for local builds
+
+```bash
+make_env.bash $TAG
```
Comment (jrhemstad) [inline docs/readme.md:10]: question: What is `$TAG`?

Comment (wmaxey) [reply docs/readme.md:10]: Literally the tag for the image that Thrust/CUB shares for building, but yeah, I'll do that.

**Review by jrhemstad — APPROVED** [review]

**Pull request merged**